include/enums.h

```diff
@@ -2748,22 +2754,134 @@ enum EnemyFlags {
+};
+
+enum EnemyActionFlags {
+    AI_ACTION_JUMP_WHEN_SEE_PLAYER          = 0x01,
+    AI_ACTION_02                            = 0x02,
+    AI_ACTION_LOOK_AROUND_DURING_LOITER     = 0x10,
+};
+
+enum PiranhaPlantStates {
+    AI_STATE_PIRANHA_PLANT_00       = 0,
+    AI_STATE_PIRANHA_PLANT_01       = 1,
+    AI_STATE_PIRANHA_PLANT_10       = 10,
+    AI_STATE_PIRANHA_PLANT_11       = 11,
+    AI_STATE_PIRANHA_PLANT_12       = 12,
+    AI_STATE_PIRANHA_PLANT_13       = 13,
+    AI_STATE_PIRANHA_PLANT_14       = 14,
+    AI_STATE_PIRANHA_PLANT_SUSPEND  = 99
+};
+
+enum AiStateNappingClubba {
+    AI_STATE_NAPPING_CLUBBA_INIT                = 0,
+    AI_STATE_NAPPING_CLUBBA_SLEEP               = 1,
+    AI_STATE_NAPPING_CLUBBA_WAKE_UP             = 2,
+    AI_STATE_NAPPING_CLUBBA_3                   = 3,
+    AI_STATE_NAPPING_CLUBBA_ACQUIRE             = 4,
+    AI_STATE_NAPPING_CLUBBA_10                  = 10,
+    AI_STATE_NAPPING_CLUBBA_11                  = 11,
+    AI_STATE_NAPPING_CLUBBA_INIT_CHASE          = AI_STATE_CHASE_INIT,
+    AI_STATE_NAPPING_CLUBBA_CHASE               = AI_STATE_CHASE,
+    AI_STATE_NAPPING_CLUBBA_LOSE_PLAYER         = AI_STATE_LOSE_PLAYER,
+    AI_STATE_NAPPING_CLUBBA_30                  = 30,
+    AI_STATE_NAPPING_CLUBBA_31                  = 31,
+    AI_STATE_NAPPING_CLUBBA_32                  = 32,
+    AI_STATE_NAPPING_CLUBBA_33                  = 33,
+    AI_STATE_NAPPING_CLUBBA_INIT_RETURN_HOME    = 40,
+    AI_STATE_NAPPING_CLUBBA_RETURN_HOME         = 41,
+    AI_STATE_NAPPING_CLUBBA_50                  = 50,
+    AI_STATE_NAPPING_CLUBBA_99                  = 99,
+};
+
+enum MeleeHitboxAttackStates {
+    MELEE_HITBOX_STATE_NONE         = 0,
+    MELEE_HITBOX_STATE_INIT         = 1,
+    MELEE_HITBOX_STATE_PRE          = 2,
+    MELEE_HITBOX_STATE_ACTIVE       = 3,  // hitbox is active
+    MELEE_HITBOX_STATE_POST         = 4
+};
+
+enum ProjectileHitboxAttackStates {
+    PROJECTILE_HITBOX_STATE_NONE        = 0,
+    PROJECTILE_HITBOX_STATE_INIT        = 1,
+    PROJECTILE_HITBOX_STATE_PRE         = 2,
+    PROJECTILE_HITBOX_STATE_ACTIVE      = 3,  // hitbox is active
+    PROJECTILE_HITBOX_STATE_POST        = 4,
+    PROJECTILE_HITBOX_STATE_DONE        = 100
+};
+
+#define VAR_PROJECTILE_HITBOX_STATE varTable[0]
+#define VAR_PROJECTILE_COUNT varTable[3]
+
+// ai script
+#define AI_TEMP_STATE                  functionTemp[0]
+#define AI_TEMP_STATE_AFTER_SUSPEND    functionTemp[1]
+// melee enemy
+#define AI_VAR_ATTACK_STATE    varTable[0]
+#define AI_VAR_MELEE_PRE_TIME  varTable[1]
+#define AI_VAR_MELEE_HIT_TIME  varTable[2]
+#define AI_VAR_MELEE_MISS_TIME varTable[3]
+#define AI_VAR_NEXT_STATE      varTable[7]
+// melee hitbox
+#define AI_VAR_HITNPC_YOFFSET  varTable[0]
+#define AI_VAR_HITNPC_DIST     varTable[1]
+#define AI_VAR_HITNPC_2        varTable[2]
+#define AI_VAR_HITNPC_3        varTable[3]
+#define AI_VAR_HITNPC_4        varTable[4]
+#define AI_VAR_HITNPC_SOUND    varTable[15]
```

Comment:
These should probably be in macros.h or a new ai.h sorta thing

Reply:
macros.h seems fine for now. I'll clean up the organization for all the AI code once they are all complete. Having them in a 'global' place like enums of macros is convenient for working with them until then.